Makefile

```
@@ -0,0 +1,53 @@
+GO_VERSION ?= 1.20.2
+GOLANG_CI_VER ?= v1.52
+GOSEC_VER ?= 2.15.0
+TEST_COVERAGE_FILE=lcov.info
+TEST_COVERAGE_HTML_FILE=coverage_unit.html
+TEST_COVERAGE_FUNC_FILE=func_coverage.out
+
+# CONTAINER_RUNNABLE checks if tests and lint check can be run inside container.
+PODMAN ?= $(shell podman -v > /dev/null 2>&1; echo $$?)
+ifeq ($(PODMAN), 0)
+CONTAINER_RUNTIME=podman
+else
+CONTAINER_RUNTIME=docker
+endif
+CONTAINER_RUNNABLE ?= $(shell $(CONTAINER_RUNTIME) -v > /dev/null 2>&1; echo $$?)
+
+.PHONY: unit_clean
+unit_clean: ## clean up the unit test artifacts created
+ifeq ($(CONTAINER_RUNNABLE), 0)
+         $(CONTAINER_RUNTIME) system prune -f
+endif
+        rm ${TEST_COVERAGE_FILE} ${TEST_COVERAGE_HTML_FILE} ${TEST_COVERAGE_FUNC_FILE} > /dev/null 2>&1
+
+.PHONY: unit
+unit: ## Run unit tests against code.
+ifeq ($(CONTAINER_RUNNABLE), 0)
+         $(CONTAINER_RUNTIME) run -it -v ${PWD}:/go/src -w /go/src docker.io/library/golang:${GO_VERSION}-alpine3.17 \
```

Line 41 doesn't specify registry as well as alpine version, so the following suggestion it's to keep image names consistent.
```suggestion
         $(CONTAINER_RUNTIME) run -it -v ${PWD}:/go/src -w /go/src golang:${GO_VERSION}-alpine \
```

---

good suggestion

---

Good suggestion